packages/grid/x-data-grid/src/tests/columnHeaders.DataGrid.test.tsx

```diff
@@ -49,4 +49,49 @@ describe('<DataGrid /> - Column Headers', () => {
       expect(getColumnHeaderCell(0)).to.have.class('foobar');
     });
   });
+
+  describe('header menu', () => {
+    it('should allow to hide column', () => {
+      const { getByRole, getAllByLabelText } = render(
+        <div style={{ width: 300, height: 300 }}>
+          <DataGrid
+            {...baselineProps}
+            columns={[{ field: 'id' }, { field: 'brand', headerClassName: 'foobar' }]}
+          />
+        </div>,
+      );
+
+      expect(getColumnHeadersTextContent()).to.deep.equal(['id', 'brand']);
+
+      fireEvent.click(getAllByLabelText('Menu')[0]);
+      fireEvent.click(getByRole('menuitem', { name: 'Hide' }));
+
+      clock.runToLast();
+      expect(getColumnHeadersTextContent()).to.deep.equal(['brand']);
+    });
+
+    it('should not allow to hide the only visible column', () => {
+      const { getByRole, getAllByLabelText } = render(
+        <div style={{ width: 300, height: 300 }}>
+          <DataGrid
+            {...baselineProps}
+            columns={[{ field: 'id' }, { field: 'brand', headerClassName: 'foobar', hide: true }]}
+          />
+        </div>,
+      );
+
+      expect(getColumnHeadersTextContent()).to.deep.equal(['id']);
+
+      fireEvent.click(getAllByLabelText('Menu')[0]);
+
+      const hideButton = getByRole('menuitem', { name: 'Hide' });
+      /**
+       * Clicking disabled `ButtonBase` as `li` element would
+       * call onClick handler in testing environment.
+       * It doesn't happen with user click in browser.
+       * So instead of firing click event, we only check `aria-disabled` here.
+       */
+      expect(hideButton.getAttribute('aria-disabled')).to.equal('true');
```

Comment:
It gives a more friendly message when it fails.

```suggestion
      expect(hideButton).to.have.attribute('aria-disabled', 'true');
```

Reply:
Huh, turns out the message is not friendly at all when it fails:
```
TypeError: Converting circular structure to JSON
            --> starting at object with constructor 'HTMLLIElement'
            |     property '__reactFiber$x0ublx5qav' -> object with constructor 'FiberNode'
            --- property 'stateNode' closes the circle
            at JSON.stringify (<anonymous>)
            at processAssertionError (node_modules/karma-mocha/lib/adapter.js:59:41)
            at Runner.<anonymous> (node_modules/karma-mocha/lib/adapter.js:164:28)
```

I tried another test in the codebase that uses this notation and made it fail - got the same output.
For example here:
https://github.com/mui-org/material-ui-x/blob/7cc6c3c68e139836fec65505254a311534f1e929/packages/grid/x-data-grid/src/tests/selection.DataGrid.test.tsx#L327

If assertion fails - you'll see `TypeError: Converting circular structure to JSON`

Looks like an issue with mocha, I'll look into it later